### PR TITLE
Worked on the pop-over of matched entity

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/dataset/shared/TagSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+type Props = {
+    urn: string;
+};
+
+export const TagSummary = ({ urn }: Props) => {
+    const { data } = useGetTagQuery({ variables: { urn } });
+    return (
+        <HoverEntityTooltip>
+            <TagLink key={data?.tag?.urn}>
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/dataset/shared/TagSummary.tsx:19]: so this should actually be a link to the tag page - so if you look at `TagTermGroup.tsx` you can export the `TagLink` component and simply import that here! you'll need to set the link like we do in that component.

File: datahub-web-react/src/app/entity/dataset/shared/TermSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+type Props = {
+    urn: string;
+};
+
+export const TermSummary = ({ urn }: Props) => {
+    const { data } = useGetGlossaryTermQuery({ variables: { urn } });
+
+    return (
+        <HoverEntityTooltip>
+            <TermLink key={data?.glossaryTerm?.urn}>
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/dataset/shared/TermSummary.tsx:21]: same thing here - replicate what we do in `TagTermGroup.tsx`

File: datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx [hunk 3]
```diff
@@ -253,6 +255,11 @@ export class DatasetEntity implements Entity<Dataset> {
     renderSearch = (result: SearchResult) => {
         const data = result.entity as Dataset;
         const genericProperties = this.getGenericEntityProperties(data);
+        const snippet = result.matchedFields[0].value.includes('urn:li:tag') ? (
+            <TagSummary urn={result.matchedFields[0].value} />
+        ) : (
+            <TermSummary urn={result.matchedFields[0].value} />
+        );
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx:262]: I don't think it's guaranteed that if this is not a tag then it is a term (in fact I don't think we surface terms here at all?)

I would say if it includes this `urn:li:tag` then use the tags summary, then do another check if it contains `urn:li:glossaryTerm` and use the terms summary, then if neither of those match, use `<b>{result.matchedFields[0].value}</b>` as it was before

Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx]: Fixed and committed!!

File: datahub-web-react/src/app/entity/dataset/shared/TermSummary.tsx (hunk 2 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/dataset/shared/TermSummary.tsx]: Fixed and committed!!

File: datahub-web-react/src/app/entity/dataset/shared/TagSummary.tsx (hunk 1 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/dataset/shared/TagSummary.tsx]: Fixed and committed!!

**Comment by jjoyce0510** [comment]
Build & test is failing!

**Review by chriscollins3456 — APPROVED** [review]
nice!

**Pull request merged**